paddle/py_paddle/dataprovider_converter.py

```diff
@@ -95,17 +98,27 @@ class DenseScanner(IScanner):
     def __init__(self, input_type, pos):
         IScanner.__init__(self, input_type, pos)
         self.__mat__ = None
+        self.__shape__ = None
         self.__height__ = 0
 
     def pre_scan(self, dat):
         self.__height__ += 1
 
-    def finish_pre_scan(self, argument):
+    def finish_pre_scan(self, argument, dat=None):
```

Comment:
Done. I move the shape settings to `pre_scan` function and also check the shape for each data.